cli/cmd/check.go

```diff
@@ -107,7 +111,7 @@ func configureAndRunChecks(w io.Writer, options *checkOptions) error {
 		RetryDeadline:         time.Now().Add(options.wait),
 	})
 
-	success := runChecks(w, hc)
+	success := runChecks(w, hc, options.output)
```

Comment:
i'm realizing that when `-o json` is set, anything that isn't json should probably be written to stderr, to allow for tools to capture strictly the json.

rough (totally untested, not compiled) code on how it could work:

```go
func configureAndRunChecks(wout io.Writer, werr io.Writer, options *checkOptions) error {
...
	success := false
	w := wout
	if options.output == jsonOutput {
		success = runChecksJSON(wout, werr, hc)
		w = werr
	} else {
		success = runChecksBasic(wout, hc)	
	}
...
	fmt.Fprintf(w, "Status check results are %s\n", okStatus)

...

func runChecksJSON(wout io.Writer, werr io.Writer, hc *healthcheck.HealthChecker) bool {
...
	resultJSON, err := json.MarshalIndent(outputJSON, "", "  ")
	if err == nil {
		fmt.Fprint(wout, string(resultJSON))
	} else {
		fmt.Fprintf(werr, "JSON serialization of the check result failed with %s", err)
	}

```

Reply:
oh you're absolutely right! Thanks!

Reply:
I've decided to implement this a bit differently, but with equal result :)